### PR TITLE
Add tests for np.row_stack

### PR DESCRIPTION
**Pull request by KrisMinchev**
Since `np.row_stack` is an alias for `np.vstack`, there is no need for a separate Numba impementation for it. This PR adds a test for `np.row_stack` for every test for `np.vstack`.

### CODE REVIEW

File: numba/tests/test_dyn_array.py [hunk 1]
```diff
@@ -34,6 +34,9 @@ def np_hstack(a, b, c):
 def np_vstack(a, b, c):
     return np.vstack((a, b, c))
 
+def np_row_stack(a,b,c):
+    return np.row_stack((a,b,c))
```
Comment (guilhermeleobas) [inline numba/tests/test_dyn_array.py:38]:
```suggestion
def np_row_stack(a, b, c):
    return np.row_stack((a, b, c))
```

File: numba/tests/test_dyn_array.py [hunk 2]
```diff
@@ -1757,15 +1760,24 @@ def test_vstack(self):
         pyfunc = np_vstack
         cfunc = nrtjit(pyfunc)
 
+        # Since np.row_stack is an alias for np.vstack, it does not need a
+        # separate Numba implementation. For every test for np.vstack, the same
+        # test for np.row_stack has been added.
+        pyfunc_rowstack = np_row_stack
+        cfunc_rowstack = nrtjit(pyfunc_rowstack)
```
Comment (guilhermeleobas) [inline numba/tests/test_dyn_array.py:1767]: Create a for loop instead:

```suggestion
        for pyfunc in (np_vstack, np_row_stack):
            cfunc = nrtjit(pyfunc)
            ...
```

**Review by guilhermeleobas — CHANGES_REQUESTED** [review]
Hi @KrisMinchev, I have only some minor changes. But PR looks great. Thanks

File: docs/upcoming_changes/9085.np_support.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Added support for np.row_stack() function.
+=============================================
+
+This PR improves on `#4074 <https://github.com/numba/numba/issues/4074>` by adding support for np.row_stack().
```
Comment (stuartarchibald) [inline docs/upcoming_changes/9085.np_support.rst:4]:
```suggestion
Support is added for ``numpy.row_stack()``.
```

File: docs/upcoming_changes/9085.np_support.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Added support for np.row_stack() function.
+=============================================
```
Comment (stuartarchibald) [inline docs/upcoming_changes/9085.np_support.rst:2]:
```suggestion
==========================================
```
think this line is too long?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, couple of minor comments else looks good.

**Comment by KrisMinchev** [comment]
> Thanks for the patch, couple of minor comments else looks good.

Hi @stuartarchibald, thank you for the comments. I have updated the towncrier accordingly.

**Review by guilhermeleobas — APPROVED** [review]
Approved! Thanks, @KrisMinchev

CI failure is not related to this PR!

**Pull request merged**